src/cache/surface_albedo.jl

```diff
@@ -0,0 +1,204 @@
+
+    # set the direct and diffuse surface albedos
+    direct_sw_surface_albedo .=
+        ones(axes(direct_sw_surface_albedo)) .* reshape(
+            surface_albedo_direct(
+                α_model,
+            ).(λ, μ, RRTMGPI.field2array(wind_speed)),
+            1,
+            :,
+        )
+    diffuse_sw_surface_albedo .=
+        ones(axes(direct_sw_surface_albedo)) .* reshape(
+            surface_albedo_diffuse(
+                α_model,
+            ).(λ, μ, RRTMGPI.field2array(wind_speed)),
+            1,
+            :,
+        )
+end
+
+"""
+    surface_albedo_direct(α_model::RegressionFunctionAlbedo{FT}) where {FT}
+
+Set the surface albedo using the regression function of Jin et al. (2011).
+"""
+function surface_albedo_direct(α_model::RegressionFunctionAlbedo{FT}) where {FT}
+    α_dir =
+        (λ, cosθ, u) -> begin
+            # relative refractive index of water and air (n = n_w/n_a)
+            n = α_model.n
+
+            # refractive index of water for visible light
+            n0 = α_model.n0
+
+            # mean wave slope distribution width
+            σ = α_model.wave_slope(u)
+
+            # Fresnel reflectance (assuming equal contribution of the p-polorized and s-polarized components, and using the perfect dielectric medium)
+            sin(cosθ) = sqrt(1 - cosθ^2)
+            rf_p(n, cosθ) =
+                (
+                    (n^2 * cosθ - sqrt(n^2 - sin(cosθ)^2)) /
+                    (n^2 * cosθ + sqrt(n^2 - sin(cosθ)^2))
+                )^2
+            rf_s(n, cosθ) =
+                (
+                    (cosθ - sqrt(n^2 - sin(cosθ)^2)) /
+                    (cosθ + sqrt(n^2 - sin(cosθ)^2))
+                )^2
+            rf(n, cosθ) = (rf_p(n, cosθ) + rf_s(n, cosθ)) / 2
+
+            # regression coefficients
+            p = α_model.p
+
+            # the regression function (J11, eq. 4)
+            f(cosθ, σ) =
+                (
+                    p[1] +
+                    p[2] * cosθ +
+                    p[3] * cosθ^2 +
+                    p[4] * cosθ^3 +
+                    p[5] * σ +
+                    p[6] * σ * cosθ
+                ) * exp(
+                    p[7] +
+                    p[8] * cosθ +
+                    p[9] * cosθ^2 +
+                    p[10] * σ +
+                    p[11] * σ * cosθ,
+                )
+
+            # return the albedo (J11, eq. 1)
+            return rf(n, cosθ) - rf(n, cosθ) / rf(n0, cosθ) * f(cosθ, σ)
+        end
+    return α_dir
+end
+
+"""
+    surface_albedo_diffuse!(α_model::RegressionFunctionAlbedo)
+
+Set the surface albedo using the Jin et al. (2011) empirical formulation.
+
+!!! note
+    For now we assume that the cloud fraction is 0.0.
+"""
+function surface_albedo_diffuse(
+    α_model::RegressionFunctionAlbedo{FT},
+) where {FT}
+    α_diff =
+        (λ, θ, u) -> begin
+            could_fraction = 0.0 # TODO: connect this to the EDMF
```

Comment:
```suggestion
            could_fraction = 0 # TODO: connect this to the EDMF
```
0.0 will be Float64? I'm not sure, @Sbozzolo?

Reply:
yes, this is safer, thanks!